src/normalizers/keywordProtector.js

```diff
@@ -0,0 +1,34 @@
+import { regexMatchingDecoderMaker } from './regex.js';
+
+export function keywordProtector (keywordMap) {
+    const encoder = regexMatchingEncoderMaker(
+        'keywordProtectorDecoder',
+        /(?<protector>protector:\w+)/g,
+        Object.fromEntries(Object.keys(keywordMap).map(kw => [ `protector:${kw}`, keywordMap[kw] ]))
+    );
+    const decoder = regexMatchingDecoderMaker(
+        'keywordProtectorDecoder',
+        new RegExp(`(?<protector>${Object.keys(keywordMap).join("|")||"NO_PROTECTED_STRINGS"})`, 'g'),
```

Comment:
do we need to support the case where the keywordMap is empty? we should throw in that case

Reply:
Ok.  Changed to throw an error message if no keywordMap provided.